ethcore/engines/validator-set/src/safe_contract.rs

```diff
@@ -283,6 +317,55 @@ impl ValidatorSafeContract {
 			Some(matched_event) => Some(SimpleList::new(matched_event.new_set))
 		}
 	}
+
+	fn filter_report_queue(&self, our_address: &Address, latest_block: BlockNumber) -> Result<(), EthcoreError>{
+		let client = self.client.read().as_ref().and_then(Weak::upgrade).ok_or("No client!")?;
+		let client = client.as_full_client().ok_or("No full client!")?;
+
+		self.queued_reports.lock().unwrap().retain(|&(malicious_validator_address, block, ref _data)| {
+			trace!(target: "engine", "Checking if report can be removed from cache");
```

Comment:
```suggestion
			trace!(target: "engine", "Checking if report should be removed from cache");
```
?
And maybe include a way to know which report we're checking here? Include the malicious validator address and the block?